src/common/commonutils/FileUtils.c

```diff
@@ -514,6 +563,288 @@ int CheckFileSystemMountingOption(const char* mountFileName, const char* mountDi
     return status;
 }
 
+static int CopyMountFile(const char* source, const char* target, void* log)
+{
+    FILE* sourceHandle = NULL;
+    FILE* targetHandle = NULL;
+    struct mntent* mountStruct = NULL;
+    int status = 0;
+
+    if ((NULL == source) || (NULL == target))
+    {
+        OsConfigLogError(log, "CopyMountFile called with invalid argument(s)");
+        return EINVAL;
+    }
+
+    if (!FileExists(source))
+    {
+        OsConfigLogInfo(log, "CopyMountFile: file '%s' not found", source);
+        return EINVAL;
+    }
+        
+    if (NULL != (targetHandle = setmntent(target, "w")))
+    {
+        if (NULL != (sourceHandle = setmntent(source, "r")))
+        {
+            while (NULL != (mountStruct = getmntent(sourceHandle)))
+            {
+                if (0 != addmntent(targetHandle, mountStruct))
+                {
+                    if (0 == (status = errno))
+                    {
+                        status = ENOENT;
+                    }
+                    
+                    OsConfigLogError(log, "CopyMountFile ('%s' to '%s'): failed adding '%s %s %s %s %d %d', addmntent() failed with %d", source, target, 
+                        mountStruct->mnt_fsname, mountStruct->mnt_dir, mountStruct->mnt_type, mountStruct->mnt_opts, mountStruct->mnt_freq, mountStruct->mnt_passno, status);
+                    break;
+                }
+            }
+        }
+        else
+        {
+            if (0 == (status = errno))
+            {
+                status = ENOENT;
+            }
+            
+            OsConfigLogError(log, "CopyMountFile: could not open source file '%s', setmntent() failed (%d)", source, status);
+        }
+        
+        fflush(targetHandle);
+        endmntent(targetHandle);
+        endmntent(sourceHandle);
+    }
+    else
+    {
+        if (0 == (status = errno))
+        {
+            status = ENOENT;
+        }
+
+        OsConfigLogError(log, "CopyMountFile: could not open target file '%s', setmntent() failed (%d)", target, status);
+    }
+        
+    return status;
+}
+
+int SetFileSystemMountingOption(const char* mountDirectory, const char* mountType, const char* desiredOption, void* log)
+{
+    const char* fsMountTable = "/etc/fstab";
+    const char* mountTable = "/etc/mtab";
+    const char tempFileNameTemplate[] = "/tmp/~xtab%d";
+    const char* newLineAsIsTemplate = "\n%s %s %s %s %d %d";
+    const char* newLineAddNewTemplate = "\n%s %s %s %s,%s %d %d";
+
+    FILE* fsMountHandle = NULL;
+    FILE* mountHandle = NULL;
+    char* newLine = NULL;
+    char* tempFileNameOne = NULL;
+    char* tempFileNameTwo = NULL;
+    struct mntent* mountStruct = NULL;
+    bool matchFound = false;
+    int lineNumber = 0;

```

Comment:
nit: Should this start at 1?

Reply:
Could. Prefer it as a 0 based index. Line 0 is first, etc.